server/src/main/java/org/opensearch/index/IndexSettings.java

```diff
@@ -618,6 +627,10 @@ private void setRetentionLeaseMillis(final TimeValue retentionLease) {
      * Is merge of flush enabled or not
      */
     private volatile boolean mergeOnFlushEnabled;
+    /**
+     * Specialized merge-on-flush policy if provided
+     */
+    private volatile Function<MergePolicy, MergePolicy> mergeOnFlushPolicy;
```

Comment:
Should we use `UnaryOperator<T>` instead

Reply:
Hm ... I think with `UnaryOperator` it is going to be less verbose, thanks!